pkg/segment/query/processor/streamstatscommand.go

```diff
@@ -18,26 +18,942 @@
 package processor
 
 import (
+	"container/list"
+	"fmt"
+	"io"
+	"math"
+	"sort"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/cespare/xxhash"
+	"github.com/siglens/siglens/pkg/common/dtypeutils"
 	"github.com/siglens/siglens/pkg/segment/query/iqr"
 	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	segutils "github.com/siglens/siglens/pkg/segment/utils"
+	putils "github.com/siglens/siglens/pkg/utils"
 )
 
 type streamstatsProcessor struct {
-	options *structs.StreamStatsOptions
+	options          *structs.StreamStatsOptions
+	currentIndex     int
+	currentBucketKey string
 }
 
 func (p *streamstatsProcessor) Process(iqr *iqr.IQR) (*iqr.IQR, error) {
-	panic("not implemented")
+	if iqr == nil {
+		return nil, io.EOF
+	}
+
+	log.Infof("streamstats: starting processing for %d records with window=%d",
+		iqr.NumberOfRecords(), p.options.Window)
+
+	outIQR := iqr.BlankCopy()
+
+	measureAggs := p.options.MeasureOperations
+	if p.options.GroupByRequest != nil {
+		measureAggs = p.options.GroupByRequest.MeasureOperations
+	}
+
+	if p.options.RunningStreamStats == nil {
+		p.options.RunningStreamStats = make(map[int]map[string]*structs.RunningStreamStatsResults)
+	}
+
+	allColumns, err := iqr.ReadAllColumns()
+	if err != nil {
+		return nil, fmt.Errorf("streamstats: failed to read columns: %v", err)
+	}
+
+	knownValues := make(map[string][]segutils.CValueEnclosure)
+
+	for colName, values := range allColumns {
+		if colName != "timestamp" { // Skip timestamp column
+			knownValues[colName] = values
+		}
+	}
+
+	for _, measureAgg := range measureAggs {
+		resultCol := measureAgg.String()
+		if _, exists := knownValues[resultCol]; !exists {
+			knownValues[resultCol] = make([]segutils.CValueEnclosure, iqr.NumberOfRecords())
+		}
+	}
+
+	bucketKey := ""
+	currentBucketKey := bucketKey
+
+	numPrevSegmentProcessedRecords := p.options.NumProcessedRecords
+	currIndex := 0
+
+	for i := 0; i < iqr.NumberOfRecords(); i++ {
+		record := make(map[string]interface{})
+		// Convert IQR columns to record format
+		for colName, values := range allColumns {
+			if i < len(values) {
+				record[colName] = values[i].CVal
+			}
+		}
+
+		if p.options.GroupByRequest != nil {
+			bucketKey = ""
+			for _, colName := range p.options.GroupByRequest.GroupByColumns {
+				if val, ok := allColumns[colName]; ok && i < len(val) {
+					bucketKey += fmt.Sprintf("%v_", val[i].CVal)
+				}
+			}
+		}
+
+		if p.options.ResetOnChange && currentBucketKey != bucketKey {
+			resetAccumulatedStreamStats(p.options)
+			currIndex = 0
+			numPrevSegmentProcessedRecords = 0
+		}
+
+		shouldResetBefore, err := evaluateResetCondition(p.options.ResetBefore, record)
+		if err != nil {
+			return nil, err
+		}
+		if shouldResetBefore {
+			resetAccumulatedStreamStats(p.options)
+			currIndex = 0
+			numPrevSegmentProcessedRecords = 0
```

Comment:
`numPrevSegmentProcessedRecords` is not required to be maintained, command should be oblivious to the concept of segments and how data is stored. You can assume that data will come in correct order. This was used earlier because in the older pipeline data use to come per segment and the ordering of records had to be done by the command itself. With the new pipeline IQR will have the correct order of records.